src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,133 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final int MIN_AGE = 18;
+    private static final int DEFAULT_AGE = 20;
+    private static final String DEFAULT_LOGIN = "admin";
+    private static final String DEFAULT_PASSWORD = "12345678";
+    private static RegistrationService registrationService;
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        user.setAge(DEFAULT_AGE);
+        user.setLogin(DEFAULT_LOGIN);
+        user.setPassword(DEFAULT_PASSWORD);
+    }
+
+    @Test
+    void validUser_Ok() {
+        assertEquals(user, registrationService.register(user));
+    }
+
+    @Test
+    void register_differentUsers_Ok() {
+        User user1 = createUser(DEFAULT_AGE, DEFAULT_PASSWORD, DEFAULT_LOGIN);
+        User user2 = createUser(DEFAULT_AGE, DEFAULT_PASSWORD, DEFAULT_LOGIN + "a");
+        registrationService.register(user);
+        registrationService.register(user2);
+        User expectedUser1 = Storage.people.get(0);
+        User expectedUser2 = Storage.people.get(1);
+        assertNotEquals(expectedUser1, expectedUser2);
+    }
+
+    @Test
+    void register_userIsNull_NotOk() {
+        checkException(null);
+    }
+
+    @Test
+    void register_NullAge_NotOk() {
```

Comment:
check case after `_`

Reply:
?

Reply:
your methods must have familiar names, if you use `register_minAge_Ok` than `register_NullAge_NotOk` must be `register_nullAge_NotOk`

Reply:
If I understand correctly, in this case I have to change the first letter after _ to a lowercase letter?